src/lock_manager.h

```diff
@@ -55,3 +58,25 @@ class LockGuard {
   LockManager *lock_mgr_ = nullptr;
   rocksdb::Slice key_;
 };
+
+class MultiLockGuard {
+ public:
+  explicit MultiLockGuard(LockManager *lock_mgr, const std::vector<std::string> &keys):
+      lock_mgr_(lock_mgr) {
+    locks_ = lock_mgr_->MultiGet(keys);
+    for (const auto &iter : locks_) {
+      iter->lock();
+    }
+  }
+
+  ~MultiLockGuard() {
+    // Lock with order `A B C` and unlock should be `C B A`
```

Comment:
why we need to unlock reversely, AFAIK, they are separate.

Reply:
Yes, for the `MultiLockGuary`, it's NOT different, but I think it's a good habit the lock and unlock with the reverse order.